src/kv/memstore.rs

```diff
@@ -117,16 +106,14 @@ impl Write for WriteTransaction<'_> {
         self
     }
     async fn put(&self, key: &str, value: &[u8]) -> Result<()> {
-        self.check_open()?;
         self.pending
             .lock()
             .await
             .insert(key.to_string(), value.to_vec());
         Ok(())
     }
 
-    async fn commit(&mut self) -> Result<()> {
-        self.check_open()?;
+    async fn commit(self: Box<Self>) -> Result<()> {
```

Comment:
did not even know you could do `self: Box<Self>`. For education, why is it important to make this method only callable on the boxed version? Like what would be wrong with putting it on &Self and then calling it through the Boxed instance.

Reply:
I don't know if this is right or not, but here's what I know: if this signature is `self: Self`, then we get `error[E0161]: cannot move a value of type dyn kv::Write: the size of dyn kv::Write cannot be statically determined` at the callers, because it's a trait. If we defined this on a reference instead it would work, but we wouldn't get the "the call ends the lifetime" behavior that we do get with this, and which seems nice for the API.

Any further thoughts?